Tests/SwiftDocCTests/Model/AvailabilityParserTests.swift

```diff
@@ -91,7 +91,7 @@ class AvailabilityParserTests: XCTestCase {
         
         /// Test all platforms
         let compiler = AvailabilityParser(availability)
-        XCTAssertTrue(compiler.isDeprecated())
+        XCTAssertFalse(compiler.isDeprecated())
```

Comment:
I'm not sure that this is the correct behavior. Previously a macOS-only target with a macOS deprecation annotation would mark a symbol as deprecated but now the same symbol would require additional annotations for platforms that the target doesn't support.

Reply:
I think DocC would need to know what the list of platforms are to determine if a given symbol is deprecated for all platforms or not. Unfortunately I don't think the information that DocC is passed in a typical build today is enough to know what all platforms are.

It's possible to provide a default availability for a platform in the documentation catalog's Info.plist file and it's also possible to provide information about the current versions of each platform via individual `--platform` arguments—but neither of those are provided by default.

Reply:
+1, can't agree anymore.

But one more question here, would this info be redundant to the Package.swift's platform params?
eg.
```swift
// Package.swift
platforms: [.iOS(.v13), .macOS(.13), .watchOS(.v5)]

// DocC Catalog's Info.plist
--platform [iOS=14, macOS=12, iOSApplicationExtension=14]
```

How should we handle such 2 info? Merge them or the second one shall prevail.
```
// Possible final platforms list for DocC to consume
1. [iOS=14,macOS=13,watchOS=5,iOSApplicationExtension=14] // Merge the 2
2. [iOS=14,macOS=13,iOSApplicationExtension=14] // Only merge when the second list has that item
3. [iOS=14, macOS=12, iOSApplicationExtension=14] // the second one shall prevail.
etc...
```

Reply:
AFAIK a Package manifests `platforms` argument defaults to the oldest deployment target version that's available in the SDK if a platforms deployment target isn't customized. See [PackageDescription/SupportedPlatform][1].

If I remove all platforms arguments except iOS from the SlothCreator sample project's Package manifest I can still build it for macOS with `swift build --triple x86_64-apple-macosx11.0`.

[1]: https://developer.apple.com/documentation/packagedescription/supportedplatform

Reply:
There is already some proposal for the platform availability issue.
https://forums.swift.org/t/setting-a-documentation-article-s-platform-availability/61144

What’s your idea on #450? @QuietMisdreavus